src/libxrpl/protocol/STTx.cpp

```diff
@@ -262,6 +265,42 @@ STTx::checkSign(
     return Unexpected("Internal signature check failure.");
 }
 
+Expected<void, std::string>
+STTx::checkBatchSign(
+    RequireFullyCanonicalSig requireCanonicalSig,
+    Rules const& rules) const
+{
+    try
+    {
+        if (getTxnType() != ttBATCH)
+        {
+            JLOG(debugLog().fatal()) << "not a batch transaction";
+            return Unexpected("Not a batch transaction.");
+        }
+        XRPL_ASSERT(
```

Comment:
Sorry, should have added this to the earlier comment - assert has to come before the if in this case, otherwise the assert will be skipped